apps/clock/js/panels/stopwatch/main.js

```diff
@@ -63,20 +71,52 @@ define(function(require) {
     element.addEventListener(
       'panel-visibilitychange', this.onvisibilitychange.bind(this));
 
+    IntlHelper.observe('timer-msS', this.updateTimeStrings.bind(this));
+
     this.setStopwatch(new Stopwatch());
 
   };
 
   Stopwatch.Panel.prototype = Object.create(Panel.prototype);
 
+  Stopwatch.Panel.prototype.updateTimeStrings = function() {
+    IntlHelper.get('timer-msS').then(formatter => {
+      var stopwatch = priv.get(this).stopwatch;
+      var e = stopwatch.getElapsedTime();
+
+      var time = formatter.format(e);
+      this.nodes.time.textContent = time;
+
+      var node = this.nodes.laps;
+      var laps = stopwatch.getLaps();
+      laps.push(stopwatch.nextLap());
+      var lapnodes = node.querySelectorAll('li.lap-cell');
+
+      var lapFormatter = IntlHelper.get('lap-number');
+
+      for (var i = 0; i < laps.length; i++) {
+        lapnodes[i].querySelector('.lap-duration').textContent =
+          formatter.format(laps[laps.length - 1 - i].duration);
+        navigator.mozL10n.setAttributes(
+          lapnodes[i].querySelector('.lap-name'),
+          'lap-number',
+          { n: lapFormatter.format(i) }
+        );
+      }
+    });
+  };
```

Comment:
I managed to trigger reorder on this retranslation. I need a better way to match laps to lapnodes.